File: math/strong_number/java/StrongNumber.java
import java.util.Scanner;

public class StrongNumber {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        while(sc.hasNextInt()){
            int a=sc.nextInt();
            System.out.println(strongNumber(a)?a+" is a strong number":a+" is NOT a strong number");

        }
    }

    private static boolean strongNumber(int number) {
        int sum=0,temp,num=number;
        while(num>0){
            temp=num%10;
            num/=10;
            sum+=factorial(temp);
        }
        return sum==number;
    }

    private static int factorial(int num) {
        for (int i = num-1; i >1; i--) {
            num*=i;
        }
        return num;
    }
}

File: math/strong_number/dart/strong_number.dart
void main() {
  isStrongNumber(145);
  isStrongNumber(0);
  isStrongNumber(40585);
  isStrongNumber(2020);
}

void isStrongNumber(int number) {
  int sum = 0;

  if (number == 0) sum = 1;

  for (int i = number; i != 0; i ~/= 10) {
    sum += factorial(i % 10);
  }

  if (sum == number)
    print("${number} is a strong number.");
  else
    print("${number} is not a strong number.");
}

int factorial(int num) {
  if (num <= 1) return 1;

  return num * factorial(num - 1);
}
